corehq/ex-submodules/couchforms/util.py

```diff
@@ -496,6 +496,10 @@ def process(xform):
                             # log the error and respond with a success code so that the phone doesn't
                             # keep trying to send the form
                             instance = _handle_known_error(e, instance)
+                            xforms[0] = instance
+                            # this is usually just one document, but if an edit errored we want
+                            # to save the deprecated form as well
+                            XFormInstance.get_db().bulk_save(xforms)
```

Comment:
before just the `instance` was saved, and the newly added deprecated form in the context just vanished.